pants-plugins/uses_services/rabbitmq_rules.py

```diff
@@ -0,0 +1,176 @@
+            return False
+        uses = target.get(UsesServicesField).value
+        return uses is not None and "rabbitmq" in uses
+
+
+@rule(
+    desc="Ensure rabbitmq is running and accessible before running tests.",
+    level=LogLevel.DEBUG,
+)
+async def rabbitmq_is_running_for_pytest(
+    request: PytestUsesRabbitMQRequest,
+) -> PytestPluginSetup:
+    # this will raise an error if rabbitmq is not running
+    _ = await Get(RabbitMQIsRunning, UsesRabbitMQRequest())
+
+    return PytestPluginSetup()
+
+
+@rule(
+    desc="Test to see if rabbitmq is running and accessible.",
+    level=LogLevel.DEBUG,
+)
+async def rabbitmq_is_running(
+    request: UsesRabbitMQRequest, platform: Platform
+) -> RabbitMQIsRunning:
+    script_path = "./is_rabbitmq_running.py"
+
+    # pants is already watching this directory as it is under a source root.
+    # So, we don't need to double watch with PathGlobs, just open it.
+    with open(is_rabbitmq_running_full_path, "rb") as script_file:
+        script_contents = script_file.read()
+
+    script_digest, kombu_pex = await MultiGet(
+        Get(Digest, CreateDigest([FileContent(script_path, script_contents)])),
+        Get(
+            VenvPex,
+            PexRequest(
+                output_filename="kombu.pex",
+                internal_only=True,
+                requirements=PexRequirements({"kombu"}),
+            ),
+        ),
+    )
+
+    result = await Get(
+        FallibleProcessResult,
+        VenvPexProcess(
+            kombu_pex,
+            argv=(
+                script_path,
+                *request.mq_urls,
+            ),
+            input_digest=script_digest,
+            description="Checking to see if RabbitMQ is up and accessible.",
+            # this can change from run to run, so don't cache results.
+            cache_scope=ProcessCacheScope.PER_SESSION,
+            level=LogLevel.DEBUG,
+        ),
+    )
+    is_running = result.exit_code == 0
+
+    if is_running:
+        return RabbitMQIsRunning()
+
+    # rabbitmq is not running, so raise an error with instructions.
+    raise ServiceMissingError.generate(
+        platform=platform,
+        messages=ServiceSpecificMessages(
+            service="rabbitmq",
+            service_start_cmd_el_7="service rabbitmq-server start",
+            service_start_cmd_el="systemctl start rabbitmq-server",
+            not_installed_clause_el="this is one way to install it:",
+            install_instructions_el=dedent(
+                """\
+                # Add key and repo for erlang and RabbitMQ
+                curl -sL https://packagecloud.io/install/repositories/rabbitmq/erlang/script.rpm.sh | sudo bash
+                curl -sL https://packagecloud.io/install/repositories/rabbitmq/rabbitmq-server/script.rpm.sh | sudo bash
+                sudo yum makecache -y --disablerepo='*' --enablerepo='rabbitmq_rabbitmq-server'
+                # Install erlang and RabbitMQ
```

Comment:
```suggestion
                # Check for any required version constraints in our docs:
                # https://docs.stackstorm.com/latest/install/rhel{platform.distro_major_version}.html

                # Install erlang and RabbitMQ (and possibly constrain the version)
```

@amanda11 wdyt? Would this work for you?

I used `latest/` in the doc string because these instructions are for st2 development.

Reply:
@cognifloyd Looked good thanks, have committed the suggestion.